src/main/java/org/apache/mesos/offer/constrain/TaskTypeGenerator.java

```diff
@@ -0,0 +1,153 @@
+package org.apache.mesos.offer.constrain;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.mesos.Protos.TaskInfo;
+import org.apache.mesos.offer.TaskException;
+import org.apache.mesos.offer.TaskUtils;
+
+
+/**
+ * This rule ensures that the given Offer is colocated with the specified task 'type', whose
```

Comment:
I think this comment is inaccurate.  Doesn't this class deal with both colocation and avoidance?